### ISSUE TITLE
Use rename instead of write and unlink when renaming entities

### ISSUE THREAD

**Issue by pofider** [jsreport/jsreport-fs-store#9]
This causes problems when storing templates in git.

**Comment by zam6ak** [comment]
I am not sure if this will work...
Any OS level file rename operation will be seen by git as 'file removed' and 'new untracked file detected'...
Unless jsreport can detect that fs is 'under' git and then use `git rename` (perhaps via option in config)

**Comment by pofider** [comment]
Hmm. I'm not a git expert. I've tried now to rename a jsreport template through studio. This causes my local git to see deleted and removed files, but if I actually commit this and push it. It is transformed in git renames. See here
https://github.com/pofider/test/commit/3e38bf3eaa74ccfe2829e28cbd9e0c3d93dc325c

Isn't this satiable enough?

**Comment by zam6ak** [comment]
I am not a git expert either, but what you experienced may have been a feature of GitHub itself....
Not sure...Maybe we need to test it after you make this change (rename instead of write and unlink)

Some more info can be found here:
http://stackoverflow.com/questions/2641146/handling-file-renames-in-git

**Comment by pofider** [comment]
Yes I've read that, but using special command sounds like quite complicated.
The commit I've linked is on the current version with unlink and write. So you can test it already on your git.

**Comment by zam6ak** [comment]
are you referring to jsreports 1.1.0 version?

**Comment by pofider** [comment]
No, this is the same for a while. See that the commit I linked is to 1.0.9 https://github.com/pofider/test/blob/master/package.json#L4

**Comment by zam6ak** [comment]
OK

I tested this, and it is not working for me the way it worked for you....
So I added entire jsreport to local git repository, added files and ran `git status` to make sure no changes...
Then, I started jsreport and ran another `git status` to double check - no changes...

In studio, then, I renamed `Sample data` to `Sample data - 2`....
Following is then `git status` output:

```
$ git status
On branch master
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        deleted:    data/data/Sample data/config.json
        deleted:    data/data/Sample data/dataJson.json

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        data/data/Sample data - 2/

no changes added to commit (use "git add" and/or "git commit -a")
```

As you can see, git did not detect rename, but rather delete and then, suggested to add these files...
This is going to be **major pain** trying to use fs-store with team of developers creating templates...

Any suggestions?

**Comment by zam6ak** [comment]
I decided to add and commit to see what happens
Here is what the log looks like (2 added, 2 deleted):

```
$ git log -1 --name-status
commit 3b15ccd3d0a01ccb290865dbcac7b01a4887b879
Author: me <my@email.com>
Date:   Wed Nov 9 15:43:19 2016 -0500

    - testing rename

A       data/data/Sample data - 2/config.json
A       data/data/Sample data - 2/dataJson.json
D       data/data/Sample data/config.json
D       data/data/Sample data/dataJson.json

```

If I 'turn on' rename detection, it shows that one file was renamed, but other was not...

```
$ git log -1 --name-status --find-renames
commit 3b15ccd3d0a01ccb290865dbcac7b01a4887b879
Author: me <my@email.com>
Date:   Wed Nov 9 15:43:19 2016 -0500

    - testing rename

A       data/data/Sample data - 2/config.json
R100    data/data/Sample data/dataJson.json     data/data/Sample data - 2/dataJson.json
D       data/data/Sample data/config.json

```

Again, this requires adding renamed files as new and it does not seem to work for all files.
